classes/webservice/WebserviceOutputJSON.php

```diff
@@ -115,19 +117,29 @@ public function renderField($field)
 
     public function renderNodeHeader($node_name, $params, $more_attr = null, $has_child = true)
     {
-        // api ?
-        static $isAPICall = false;
-        if ($node_name == 'api' && ($isAPICall == false)) {
-            $isAPICall = true;
+        if (in_array($node_name, ['description', 'schema', 'api'])) {
+            return;
```

Comment:
I think this method should return a string (even if it's an empty string)
```suggestion
            return '';
```

Reply:
Why would it need to return string? This function actually doesnt need to return anything - unless its specified by some parent function (which is not, at least in current version). The only purpose of this function is to add node to `$this->content` property - or set node as parent. Generating output string is being done further, by calling `json_encode` on `$this->content` property, after it gets fully composed.

Its sister's function in _WebserviceOutputXML_ should return string, as rendering of XML is done using concatenating strings (which is in my opinion should also be rather switched to composing XML using SimpleXML object), but who has time to rewrite everything, eh? :)

Reply:
It would need to return a string because as you said, `WebserviceOutputXML` is returning string and both `WebserviceOutputXML` & `WebserviceOutputJSON` are implementing the interface `WebserviceOutputInterface`. Then in `WebserviceOutputBuilder` we are using the interface and we expect the method `renderNodeHeader` to return string (like [here](https://github.com/PrestaShop/PrestaShop/blob/develop/classes/webservice/WebserviceOutputBuilder.php#L317) for instance).
More globally, it make sense that classes implementing the same interface return the same type of value.

Reply:
Yeah, youre totally right, I missed there is an Interface